openreview/conference/templates/programchairWebfield.js

```diff
@@ -999,4 +1124,189 @@ $('#group-container').on('click', 'a.collapse-btn', function(e) {
   return false;
 });
 
+$('#group-container').on('click', 'button.btn.btn-assign-reviewer', function(e) {
+  var $link = $(this);
+  var paperNumber = $link.data('paperNumber');
+  var paperForum = $link.data('paperForum');
+  var $currDiv = $('#' + paperNumber + '-add-reviewer');
+  var userToAdd = $currDiv.find('input').attr('value_id').trim();
+
+  if (!userToAdd) {
+    promptError('Please enter a valid email for assigning a reviewer');
+    return false;
+  }
+  var alreadyAssigned = _.find(reviewerSummaryMap[paperNumber].reviewers, function(rev){
+    return (rev.email === userToAdd) || (rev.id === userToAdd);
+  });
+  if (alreadyAssigned) {
+    promptError('Reviewer ' + view.prettyId(userToAdd) + ' has already been assigned to Paper ' + paperNumber.toString());
+    return false;
+  }
+
+  var nextAnonNumber = findNextAnonGroupNumber(paperNumber);
+  Webfield.put('/groups/members', {
+    id: CONFERENCE_ID + '/Paper' + paperNumber + '/Reviewers',
+    members: [userToAdd]
+  })
+  .then(function(result) {
+    var commonReaders = [CONFERENCE_ID + '/Program_Chairs'];
+    if (SHOW_AC_TAB){
+      commonReaders.push(CONFERENCE_ID + '/Paper' + paperNumber + '/Area_Chairs');
+    }
+    return Webfield.post('/groups', {
+      id: CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber,
+      members: [userToAdd],
+      readers: commonReaders.concat(CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber),
+      nonreaders: [CONFERENCE_ID + '/Paper' + paperNumber + '/Authors'],
+      writers: commonReaders,
+      signatures: [CONFERENCE_ID + '/Program_Chairs'],
+      signatories: [CONFERENCE_ID + '/Paper' + paperNumber + '/AnonReviewer' + nextAnonNumber]
+    })
+  })
+  .then(function(result) {
+    return Webfield.put('/groups/members', {
+      id: REVIEWERS_ID,
+      members: [userToAdd]
+    })
+  })
+  .then(function(result) {
+    return getUserProfiles([userToAdd]);
```

Comment:
Could we try to get the profile first and then add the user to the other groups? I think using the profile id in all the operations can be better, right?

Reply:
Just made this change and it works perfectly.